### PR TITLE
Tests: overhaul to eliminate require(mock...) from source. Adds browserify and webpack tests

### PR DESCRIPTION
**Pull request by rwaldron**
Signed-off-by: Rick Waldron <waldron.rick@gmail.com>

### CODE REVIEW

**Comment by rwaldron** [comment]
@monteslu @bcjordan can either or both try this? 

```
npm install rwaldron/firmata.js#117
```

**Comment by bcjordan** [comment]
:clap: :tada:! Works in both the test firmata-webpack-test and firmata-browserify-test repos. Awesome testing.

(Looks like s/`firmata.js`/`firmata` in fork repo name)

Ran:

```
npm install rwaldron/firmata#117
npm run build
```

Confirmed that it also makes an `npm link`-ed version of `johnny-five` browserify properly. @ ping me when this hits `johnny-five`, would love to ship a demo using this today/tomorrow.

**Comment by monteslu** [comment]
also verified working with webpack.

File: test/unit/bundling.test.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  it("must browserify", function(done) {
+    this.timeout(2500);
+    var b = browserify(entry);
+
+    b.bundle(function(error, buffer) {
+      var bundle = buffer.toString();
+      assert.equal(error, null);
+      lines.slice(startAt).forEach(function(line) {
+        assert.equal(bundle.includes(line), true);
```
Comment (rwaldron) [inline test/unit/bundling.test.js]: I wanted to concretely assert that the source ends up in the bundle... Overkill? Maybe.

Comment (bcjordan) [inline test/unit/bundling.test.js]: I like it

**Comment by rwaldron** [comment]
@bcjordan sorry I missed that demo window :( I lost track of this while working on several blog posts. In the future, please ping my by any means necessary

**Pull request merged**